uwuno/src/main/java/com/learning/uwuno/cards/basicCard.java

```diff
@@ -14,4 +14,29 @@ public basicCard(int value, Color color) {
     public int getValue() {
         return value;
     }
+
+    // Override Functions
+    // Overrides equal to compare CardType, Colour and CardValue
+    @Override
+    public boolean equals(Object obj) {
+        // Default checks
+        if (this == obj)
+            return true;
+        else if (obj == null || getClass() != obj.getClass())
+            return false;
+        // Extra checks, need cardType, cardColor and cardValue to be the same
+        basicCard toTest = (basicCard) obj;
+        return this.cardType == toTest.getType() && this.color == toTest.getColor() && this.value == toTest.getValue();
+    }
+
+    @Override
+    public int hashCode() {
```

Comment:
Can you explain these numbers, are they just random hard coded values?

Reply:
Yes, hashCode usually use odd numbers. I just used the one from Baeldung.